### DISCUSSION TITLE
Export failed with errors + all symbols get detached

### DISCUSSION THREAD

**Discussion by cargeo** [ingrammicro/puzzle-publisher#82] [General]
When exporting one or multiple artboards via "Export selected to HTML", I get the following error:

`
TypeError: MSSwatchProvider.alloc().initWithDocument is not a function. (In 'MSSwatchProvider.alloc().initWithDocument(n._getMSDocument())', 'MSSwatchProvider.alloc().initWithDocument' is undefined)
`

Additionally - and that is the worst part - all symbols of the selected artboards get detached 😱

I tried to narrow it down and made a test-file (attached). It seems the bug only occurs when artboards have a symbol.
[test.zip](https://github.com/ingrammicro/puzzle-publisher/files/9371865/test.zip)

I already ran into this in previous plugin versions, but nonetheless the export worked.

**Comment by MaxBazarov** [comment]
Thanks for the report. Will fix it ASAP.

**Comment by MaxBazarov** [comment]
Fixed in 17.8.1
https://github.com/ingrammicro/puzzle-publisher/raw/master/PuzzlePublisher.sketchplugin.zip

**Reply by cargeo** [reply]
Thanks for your quick reply and fix! 🙏

Meanwhile I had installed an older Sketch version (89) and had no issues exporting artboards to HTML. But glad plugin is now working  again on latest Sketch. 👍
